### ISSUE TITLE
Timeout crasher f672

### ISSUE THREAD

**Issue by meheff** [google/xls#686]
Running opt_main on the following takes a loooong time:

```package sample

file_number 0 "fake_file.x"

fn __sample__x14(x15: bits[7]) -> ((bits[7]), bits[7], ((bits[7]), bits[7])) {
  x16: (bits[7]) = tuple(x15, id=2, pos=[(0,6,19)])
  x17: ((bits[7]), bits[7]) = tuple(x16, x15, id=3, pos=[(0,7,25)])
  ret tuple.4: ((bits[7]), bits[7], ((bits[7]), bits[7])) = tuple(x16, x15, x17, id=4, pos=[(0,8,2)])
}

top fn __sample__main(x0: bits[7], x1: bits[37], x2: bits[57], x3: bits[745]) -> (bits[30], bits[796], bits[1490], bits[7][6], bits[1490]) {
  x6: bits[1490] = concat(x3, x3, id=10, pos=[(0,12,26)])
  one_hot.15: bits[1491] = one_hot(x6, lsb_prio=true, id=15, pos=[(0,14,24)])
  x5: bits[7][1] = array(x0, id=9, pos=[(0,11,18)])
  literal.11: bits[64] = literal(value=0, id=11)
  encode.16: bits[11] = encode(one_hot.15, id=16)
  array_index.12: bits[7] = array_index(x5, indices=[literal.11], id=12, pos=[(0,13,22)])
  array_index.13: bits[7] = array_index(x5, indices=[literal.11], id=13, pos=[(0,13,22)])
  x8: bits[1490] = zero_ext(encode.16, new_bit_count=1490, id=17, pos=[(0,14,24)])
  x7: bits[1] = eq(array_index.12, array_index.13, id=14, pos=[(0,13,22)])
  literal.19: bits[32] = literal(value=0, id=19, pos=[(0,16,49)])
  sign_ext.23: bits[1490] = sign_ext(x2, new_bit_count=1490, id=23)
  bit_slice.28: bits[745] = bit_slice(x3, start=0, width=745, id=28)
  x9: bits[796] = bit_slice(x8, start=0, width=796, id=18, pos=[(0,15,24)])
  zero_ext.30: bits[7] = zero_ext(x7, new_bit_count=7, id=30)
  array_index.20: bits[7] = array_index(x5, indices=[literal.19], id=20, pos=[(0,16,44)])
  x11: bits[1490] = add(sign_ext.23, x8, id=24, pos=[(0,17,43)])
  sign_ext.32: bits[1490] = sign_ext(x2, new_bit_count=1490, id=32)
  x12: bits[1490] = neg(x6, id=25, pos=[(0,18,22)])
  x21: bits[745][1] = array(bit_slice.28, id=29, pos=[(0,21,27)])
  x19: ((bits[7]), bits[7], ((bits[7]), bits[7]))[1] = map(x5, to_apply=__sample__x14, id=27, pos=[(0,20,28)])
  zero_ext.37: bits[1490] = zero_ext(x9, new_bit_count=1490, id=37)
  x22: bits[7] = add(x0, zero_ext.30, id=31, pos=[(0,22,21)])
  x25: bits[1] = one_hot_sel(x7, cases=[x7], id=36, pos=[(0,25,29)])
  x31: bits[30] = literal(value=48984791, id=42, pos=[(0,30,21)])
  x10: bits[7][6] = array_slice(x5, x6, width=6, id=22, pos=[(0,16,24)])
  array.21: bits[7][6] = array(array_index.20, array_index.20, array_index.20, array_index.20, array_index.20, array_index.20, id=21, pos=[(0,16,39)])
  x13: bits[1148] = dynamic_bit_slice(x11, x9, width=1148, id=26, pos=[(0,19,27)])
  x23: bits[1490] = or(x11, sign_ext.32, id=33, pos=[(0,23,28)])
  x24: (bits[7][1], bits[1490], bits[745][1], ((bits[7]), bits[7], ((bits[7]), bits[7]))[1], bits[7]) = tuple(x5, x12, x21, x19, x0, id=34, pos=[(0,24,61)])
  array.35: bits[1][1] = array(x7, id=35, pos=[(0,25,34)])
  x26: bits[1490] = xor(zero_ext.37, x12, id=38, pos=[(0,26,43)])
  x27: bits[7] = not(x22, id=39, pos=[(0,27,16)])
  x29: bits[7][1][1] = array(x5, id=40, pos=[(0,28,20)])
  x30: bits[1] = and_reduce(x25, id=41)
  x32: bits[223] = bit_slice(x8, start=1267, width=223, id=43, pos=[(0,31,25)])
  ret tuple.44: (bits[30], bits[796], bits[1490], bits[7][6], bits[1490]) = tuple(x31, x9, x6, x10, x8, id=44, pos=[(0,32,2)])
}
```

**Comment by meheff** [comment]
Fixed by d3fd8a5

**Comment by allight** [comment]
Seems to now cause an (unrelated?) crash in llvm!
